### ISSUE TITLE
Possible Plex improvements

### ISSUE THREAD

**Issue by Hellowlol** [Hellowlol/HTPC-Manager#144]
This is just ideas prs welcome :)

- [x] add filter/search
- [x] Add sort
- [x] pause button sends play, js sets wrong data attr
- [x] remove hardcoded video in playerque (cant start music playback)
- [ ] modals for music

**Comment by Hellowlol** [comment]
@raysguy @d2dyno, @thejrose1984, @GTunney, @brendann993 I was thinking about adding sort/filter to the plex module.

The idea is to reuse the search field:

So say you are on the movie page: and write 16 blocks, it will filter out everything else. I would like to add some more advanced stuff as well so. Say you want all movies with released in year 2000 you could write year: 2000

Would this be something your interested in or am i just waisting my time?

**Comment by thejrose1984** [comment]
Seems like a good concept to put into play when you have time :)

**Comment by brendann993** [comment]
great idea, definitely worth it when you have time(I know I would use it :) )

**Comment by thatbritguy** [comment]
I think that would be quite a useful feature, especially for those with larger libraries.  Will the initial load limit of 50 items restrict the ability to search though?

**Comment by Hellowlol** [comment]
No, actually everything is loaded at the same time. Its just the response that are reduced to 50:)

**Comment by thatbritguy** [comment]
Awesome :)

**Comment by Hellowlol** [comment]
@raysguy @d2dyno, @thejrose1984, @GTunney, @brendann993 can you test the lastest update?

You will search on what ever tab is open, so say your in the movie tab and search for 16 blocks, you will find it.

Examples:
year<=2000
genre=action
title=16 blocks
title=s&genre=action

**Comment by thatbritguy** [comment]
Just downloaded the new version and restarted it but the search/filter functionality isn't working at all for me.  When I search it just performs a newznab search.  Tried disabling newznab but now when I try to search nothing happens

**Comment by Hellowlol** [comment]
@raysguy. You cleared browser cache first?

**Comment by brendann993** [comment]
Thanks, It works pretty good, I haven't been able to get genre search to work. This is what appears in the log each search but nothing is actually displayed. (I do have movies which are action)

```
2015-07-04 20:55:31 modules.plex    DEBUG   _filter response was all?genre=235
2015-07-04 20:55:31 modules.plex    DEBUG   called _filter genre=action
2015-07-04 20:55:31 modules.plex    DEBUG   Fetching Movies
```

**Comment by Hellowlol** [comment]
And you had the movies tab open? Can I have the full log please?

**Comment by brendann993** [comment]
Yea movie tab open 
http://pastebin.com/VFcSrd1j
full log

**Comment by thatbritguy** [comment]
Oops, I did not!

Tested it quite a bit on both Safari and Chrome and finding it works intermittently.  The first search is usually fine, but deleting the contents of the search box and typing a new search often doesn't work.  It works a bit more consistently if you prefix the search with "title=" and just delete the actual name.  The same thing happens with other types of search.  Seems to work better if you only delete and change the last character or two.  E.g. "year=2015" shows me films for that year but if I delete that and type in "year=2014" it doesn't show any results.  If after an initial search for "year=2015" I just hit backspace on the 5 to change it to "year=2014" then it works.

The other thing I've noticed is that often it seems to search only the first couple of letters of the title.  E.g. typing "Brave" gave me three results - Brave, Brother Bear and Twilight Breaking Dawn.  This happens with or without the "title=" prefix.  Typing "title=twilight" correctly returned all the Twilight movies but also included You Only Live Twice, so again I think it was just searching the first couple of letters.

Here are the full logs for my latest test session: http://pastebin.com/3e8muN1V :)
